SEP-0012.md

```diff
@@ -0,0 +1,495 @@
+
+        When sliced with a string it should return a new ExtraCoords object
+        with only those coordinates with the given names. When sliced with a
+        numpy array like slice it should return a new ExtraCoords with the
+        slice applied. Supporting step is not required and "fancy indexing" is
+        not supported.
+        """
+```
+
+## Global Coords
+
+The final part of the `NDCube` API is a class to provide structured metadata for coordinate values which are applicable to the whole cube.
+One possible use (not mandated by this SEP) is to have any coordinates dropped by a slicing operation on a higher dimensionality cube be tracked here.
+This class is only a slight extension of the API for an immutable dictionary, to enable the tracking of the physical type of the coordinate as well as it's name and value.
+The class behaves like a mapping between coordinate name and coordinate object for all read operations.
+
+
+```python
+
+from collections.abc import Mapping
+
+
+class GlobalCoordsABC(Mapping):
+    """
+    A structured representation of coordinate information applicable to a whole NDCube.
+
+    This class acts as a mapping between coordinate name and the coordinate object.
+    In addition to this a physical type is stored for each coordinate name.
+    A concrete implementation of this class must furfill the `Mapping` ABC,
+    including methods such as `__iter__` and `__len__`.
+
+    Parameters
+    ----------
+    ndcube : `.NDCube`, optional
+        The parent ndcube for this object. Used to extract global coordinates
+        from the wcs and extra coords of the ndcube. If not specified only
+        coordinates explicitly added will be shown.
+    """
+    def __init__(self, ndcube: Optional[NDCubeABC] = None):
+        pass
+
+    def add(self,
+            name: str,
+            physical_type: str,
+            coord: Any
+            ):
+        """
+        Add a new coordinate to the collection.
+
+        Parameters
+        ----------
+        name
+            The name for the coordinate.
+        physical_type
+            An IOVA UCD1+ physical type description for the coordinate
+            (http://www.ivoa.net/documents/latest/UCDlist.html). If no matching UCD
+            type exists, this can instead be ``"custom:xxx"``, where ``xxx`` is an
+            arbitrary string. If not known, can be `None`.
+        coord
+            The object describing the coordinate value, for example a
+            `~astropy.units.Quantity` or a `~astropy.coordinates.SkyCoord`.
+        """
+
+    def remove(self, name: str):
+        """
+        Remove a coordinate from the collection.
+        """
+
+    @property
+    def physical_types(self):
+        """
+        A mapping of names to physical types for each coordinate.
+        """
+
+    def __getitem__(self, item: str):
+        """
+        Indexing the object by name should return the coordinate object.
+        """
+```
```

Comment:
A section that describes edge cases and errors would be useful to gather feedback on as well and should be added here. For example, what if a user crops to values that are not in an image? Should the result be an error or an empty array? People would likely expect certain behaviors based on their intuition and it would be good to gather that feedback.

Reply:
Is this comment aimed at just this `GlobalCoords` class or more general with NDCube and both this and `ExtraCoords`?